### ISSUE TITLE
Python lib, segmentation fault on collection creation/switch

### ISSUE THREAD

**Issue by hazelnutfox** [vincentdchan/PoloDB#38]
Python 3.9.7 (both in virtual environment and global install), polodb version 0.10.0. Ubuntu 18.04.

Fresh database (created for testing, I haven't used polo before)

```py
 ➜ python
Python 3.9.7 (default, Sep 10 2021, 00:03:59) 
[GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import polodb
>>> db = polodb.Database("testpolo")
>>> db.createCollection("testing")
Segmentation fault (core dumped)
```
Despite the segfault, creation is successful:
```py
 ➜ python
>>> import polodb
>>> db = polodb.Database("testpolo")
>>> db.createCollection("testing")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
Exception: collection name 'testing' already exists
>>> db.collection("testing")
Segmentation fault (core dumped)
```
but I cannot switch to the collection anyway because segfault again.

Edit: Same thing with same usage in lower version (0.8.1 from pypi, although polodb.version() says it's 0.8.0?)

**Comment by PythonTryHard** [comment]
Reproducible in KDE neon 5.22.5 (Ubuntu 20.04). Segfaulted as well. [Core dump](https://github.com/vincentdchan/PoloDB/files/7243937/core_dump.zip) is attached.
```python
╰─❯ python            
Python 3.9.6 (default, Jul  3 2021, 01:12:59) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import polodb
>>> polodb.version()
'0.10.0'

db = polodb.Database("testpolo")
db.createCollection("testing")
[1]    1485732 segmentation fault  python
```

**Comment by uncomfyhalomacro** [comment]
Reproducible in Void Linux.
```py
❯ python3
Python 3.9.7 (default, Aug 31 2021, 13:27:55) 
[GCC 10.2.1 20201203] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> db = polodb.Database("testpolo")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'polodb' is not defined
>>> import polodb
>>> db = polodb.Database("testpolo")
>>> db.createCollection("testing")
zsh: segmentation fault  python3
```

```py 
❯ python3
Python 3.9.7 (default, Aug 31 2021, 13:27:55) 
[GCC 10.2.1 20201203] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import polodb
>>> db = polodb.Database("testpolo")
>>> db.createCollection("testing")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
Exception: collection name 'testing' already exists
>>> db.collection("testing")
zsh: segmentation fault  python3
```

**Comment by hazelnutfox** [comment]
Got some people from PyDis (hi, @PythonTryHard ~) to test it for me and it seems it breaks in different linux distros (Ubuntu 18.04, Ubuntu 20.04, some Fedora, Void Linux...) but it works on Windows.

**Comment by vincentdchan** [comment]
Thanks for the report, I have reproduced the error. The crash occurs in the binding extension code, not in the polodb_core, it's not hard to fix. Will be fixed later.
```
#0  0x00007f1b0c334703 in CollectionObject_init (self=0x7f1b0c5dd5a0, args=<optimized out>, 
    kwds=<optimized out>) at polodb_ext.c:275
275	  if (Py_TYPE(db_obj) != &DatabaseObjectType) {
```
> Same thing with same usage in lower version (0.8.1 from pypi, although polodb.version() says it's 0.8.0?)

@hazelnutfox  `polodb.version()` returns the version of polodb_core written in Rust. It's not equal to the Python version. If you have a better idea, please tell me.

**Comment by vincentdchan** [comment]
By the way, I want to remove `createCollection` in the future. If the collection  doesn't exist, `collection` method will create a new one. What's your idea?

**Comment by vincentdchan** [comment]
A have published a new version to PyPI: `polodb-0.10.1`, fixed this issue.

**Comment by PythonTryHard** [comment]
Confirmed fixed on Python 3.9.7 (Ubuntun 21.04).
